--[=[
 d888b  db    db d888888b      .d888b.      db      db    db  .d8b.  
88' Y8b 88    88   `88'        VP  `8D      88      88    88 d8' `8b 
88      88    88    88            odD'      88      88    88 88ooo88 
88  ooo 88    88    88          .88'        88      88    88 88~~~88 
88. ~8~ 88b  d88   .88.        j88.         88booo. 88b  d88 88   88    @uniquadev
 Y888P  ~Y8888P' Y888888P      888888D      Y88888P ~Y8888P' YP   YP  CONVERTER 

designed using localmaze gui creator
]=]

-- Instances: 52 | Scripts: 0 | Modules: 0 | Tags: 0
local LMG2L = {};

-- Players.giruoh_jgry.PlayerGui.ScreenGui
LMG2L["ScreenGui_1"] = Instance.new("ScreenGui", game:GetService("Players").LocalPlayer:WaitForChild("PlayerGui"));
LMG2L["ScreenGui_1"]["ZIndexBehavior"] = Enum.ZIndexBehavior.Sibling;


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key
LMG2L["Frame key_2"] = Instance.new("Frame", LMG2L["ScreenGui_1"]);
LMG2L["Frame key_2"]["BorderSizePixel"] = 0;
LMG2L["Frame key_2"]["BackgroundColor3"] = Color3.fromRGB(89, 89, 89);
LMG2L["Frame key_2"]["Size"] = UDim2.new(0, 500, 0, 352);
LMG2L["Frame key_2"]["Position"] = UDim2.new(0, 6, 0, 0);
LMG2L["Frame key_2"]["Name"] = [[Frame key]];


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.ImageLabel
LMG2L["ImageLabel_3"] = Instance.new("ImageLabel", LMG2L["Frame key_2"]);
LMG2L["ImageLabel_3"]["BorderSizePixel"] = 0;
LMG2L["ImageLabel_3"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
-- LMG2L["ImageLabel_3"]["ImageContent"] = ;
LMG2L["ImageLabel_3"]["Image"] = [[rbxassetid://84478430508846]];
LMG2L["ImageLabel_3"]["Size"] = UDim2.new(0, 100, 0, 96);
LMG2L["ImageLabel_3"]["BackgroundTransparency"] = 1;
LMG2L["ImageLabel_3"]["Position"] = UDim2.new(0, 48, 0, 56);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.TextLabel
LMG2L["TextLabel_4"] = Instance.new("TextLabel", LMG2L["Frame key_2"]);
LMG2L["TextLabel_4"]["TextWrapped"] = true;
LMG2L["TextLabel_4"]["BorderSizePixel"] = 0;
LMG2L["TextLabel_4"]["TextScaled"] = true;
LMG2L["TextLabel_4"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextLabel_4"]["FontFace"] = Font.new([[rbxasset://fonts/families/Inconsolata.json]], Enum.FontWeight.Regular, Enum.FontStyle.Normal);
LMG2L["TextLabel_4"]["TextColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextLabel_4"]["BackgroundTransparency"] = 1;
LMG2L["TextLabel_4"]["Size"] = UDim2.new(0, 326, 0, 48);
LMG2L["TextLabel_4"]["Text"] = [[Welcome back.]];
LMG2L["TextLabel_4"]["Position"] = UDim2.new(0, 138, 0, 90);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.TextBox
LMG2L["TextBox_5"] = Instance.new("TextBox", LMG2L["Frame key_2"]);
LMG2L["TextBox_5"]["BorderSizePixel"] = 0;
LMG2L["TextBox_5"]["TextWrapped"] = true;
LMG2L["TextBox_5"]["TextSize"] = 9;
LMG2L["TextBox_5"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextBox_5"]["RichText"] = true;
LMG2L["TextBox_5"]["MultiLine"] = true;
LMG2L["TextBox_5"]["ClearTextOnFocus"] = false;
LMG2L["TextBox_5"]["PlaceholderText"] = [[enter the key]];
LMG2L["TextBox_5"]["Size"] = UDim2.new(0, 316, 0, 28);
LMG2L["TextBox_5"]["Position"] = UDim2.new(0, 98, 0, 146);
LMG2L["TextBox_5"]["Text"] = [[]];


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.TextButton
LMG2L["TextButton_6"] = Instance.new("TextButton", LMG2L["Frame key_2"]);
LMG2L["TextButton_6"]["TextWrapped"] = true;
LMG2L["TextButton_6"]["BorderSizePixel"] = 0;
LMG2L["TextButton_6"]["TextScaled"] = true;
LMG2L["TextButton_6"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextButton_6"]["Size"] = UDim2.new(0, 154, 0, 36);
LMG2L["TextButton_6"]["Text"] = [[enter]];
LMG2L["TextButton_6"]["Position"] = UDim2.new(0, 98, 0, 180);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.TextButton
LMG2L["TextButton_7"] = Instance.new("TextButton", LMG2L["Frame key_2"]);
LMG2L["TextButton_7"]["TextWrapped"] = true;
LMG2L["TextButton_7"]["BorderSizePixel"] = 0;
LMG2L["TextButton_7"]["TextScaled"] = true;
LMG2L["TextButton_7"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextButton_7"]["Size"] = UDim2.new(0, 154, 0, 36);
LMG2L["TextButton_7"]["Text"] = [[clear]];
LMG2L["TextButton_7"]["Position"] = UDim2.new(0, 260, 0, 180);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.TextLabel
LMG2L["TextLabel_8"] = Instance.new("TextLabel", LMG2L["Frame key_2"]);
LMG2L["TextLabel_8"]["BorderSizePixel"] = 0;
LMG2L["TextLabel_8"]["TextXAlignment"] = Enum.TextXAlignment.Left;
LMG2L["TextLabel_8"]["TextYAlignment"] = Enum.TextYAlignment.Bottom;
LMG2L["TextLabel_8"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextLabel_8"]["BackgroundTransparency"] = 1;
LMG2L["TextLabel_8"]["Size"] = UDim2.new(0, 492, 0, 26);
LMG2L["TextLabel_8"]["Text"] = [[00script: is made by 00script. made in Thailand bro]];
LMG2L["TextLabel_8"]["Position"] = UDim2.new(0, 4, 0, 322);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.UIDragDetector
LMG2L["UIDragDetector_9"] = Instance.new("UIDragDetector", LMG2L["Frame key_2"]);



-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.ImageLabel
LMG2L["ImageLabel_a"] = Instance.new("ImageLabel", LMG2L["Frame key_2"]);
LMG2L["ImageLabel_a"]["BorderSizePixel"] = 0;
LMG2L["ImageLabel_a"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
-- LMG2L["ImageLabel_a"]["ImageContent"] = ;
LMG2L["ImageLabel_a"]["Image"] = [[rbxassetid://84478430508846]];
LMG2L["ImageLabel_a"]["Size"] = UDim2.new(0, 26, 0, 24);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.TextLabel
LMG2L["TextLabel_b"] = Instance.new("TextLabel", LMG2L["Frame key_2"]);
LMG2L["TextLabel_b"]["BorderSizePixel"] = 0;
LMG2L["TextLabel_b"]["TextSize"] = 12;
LMG2L["TextLabel_b"]["TextXAlignment"] = Enum.TextXAlignment.Left;
LMG2L["TextLabel_b"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextLabel_b"]["Size"] = UDim2.new(0, 476, 0, 24);
LMG2L["TextLabel_b"]["Text"] = [[Project Forgot the name version 2 ]];
LMG2L["TextLabel_b"]["Position"] = UDim2.new(0, 24, 0, 0);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame
LMG2L["Frame_c"] = Instance.new("Frame", LMG2L["Frame key_2"]);
LMG2L["Frame_c"]["Visible"] = false;
LMG2L["Frame_c"]["BorderSizePixel"] = 0;
LMG2L["Frame_c"]["BackgroundColor3"] = Color3.fromRGB(89, 89, 89);
LMG2L["Frame_c"]["Size"] = UDim2.new(0, 500, 0, 352);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.ImageLabel
LMG2L["ImageLabel_d"] = Instance.new("ImageLabel", LMG2L["Frame_c"]);
LMG2L["ImageLabel_d"]["BorderSizePixel"] = 0;
LMG2L["ImageLabel_d"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
-- LMG2L["ImageLabel_d"]["ImageContent"] = ;
LMG2L["ImageLabel_d"]["Image"] = [[rbxassetid://84478430508846]];
LMG2L["ImageLabel_d"]["Size"] = UDim2.new(0, 26, 0, 24);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextLabel
LMG2L["TextLabel_e"] = Instance.new("TextLabel", LMG2L["Frame_c"]);
LMG2L["TextLabel_e"]["BorderSizePixel"] = 0;
LMG2L["TextLabel_e"]["TextSize"] = 12;
LMG2L["TextLabel_e"]["TextXAlignment"] = Enum.TextXAlignment.Left;
LMG2L["TextLabel_e"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextLabel_e"]["Size"] = UDim2.new(0, 476, 0, 24);
LMG2L["TextLabel_e"]["Text"] = [[Project Forgot the name version 2 ]];
LMG2L["TextLabel_e"]["Position"] = UDim2.new(0, 24, 0, 0);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextButton
LMG2L["TextButton_f"] = Instance.new("TextButton", LMG2L["Frame_c"]);
LMG2L["TextButton_f"]["BorderSizePixel"] = 2;
LMG2L["TextButton_f"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextButton_f"]["Size"] = UDim2.new(0, 100, 0, 20);
LMG2L["TextButton_f"]["BorderColor3"] = Color3.fromRGB(103, 103, 103);
LMG2L["TextButton_f"]["Text"] = [[execute]];
LMG2L["TextButton_f"]["Position"] = UDim2.new(0, 4, 0, 28);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextButton
LMG2L["TextButton_10"] = Instance.new("TextButton", LMG2L["Frame_c"]);
LMG2L["TextButton_10"]["BorderSizePixel"] = 2;
LMG2L["TextButton_10"]["BackgroundColor3"] = Color3.fromRGB(174, 174, 174);
LMG2L["TextButton_10"]["Size"] = UDim2.new(0, 100, 0, 20);
LMG2L["TextButton_10"]["BorderColor3"] = Color3.fromRGB(104, 104, 104);
LMG2L["TextButton_10"]["Text"] = [[setting]];
LMG2L["TextButton_10"]["Position"] = UDim2.new(0, 110, 0, 28);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.ScrollingFrame
LMG2L["ScrollingFrame_11"] = Instance.new("ScrollingFrame", LMG2L["Frame_c"]);
LMG2L["ScrollingFrame_11"]["ScrollingDirection"] = Enum.ScrollingDirection.Y;
LMG2L["ScrollingFrame_11"]["BorderSizePixel"] = 0;
LMG2L["ScrollingFrame_11"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["ScrollingFrame_11"]["VerticalScrollBarPosition"] = Enum.VerticalScrollBarPosition.Left;
LMG2L["ScrollingFrame_11"]["HorizontalScrollBarInset"] = Enum.ScrollBarInset.ScrollBar;
LMG2L["ScrollingFrame_11"]["Size"] = UDim2.new(0, 334, 0, 208);
LMG2L["ScrollingFrame_11"]["ScrollBarImageColor3"] = Color3.fromRGB(0, 0, 0);
LMG2L["ScrollingFrame_11"]["Position"] = UDim2.new(0, 4, 0, 52);
LMG2L["ScrollingFrame_11"]["LayoutOrder"] = 1;


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.ScrollingFrame.TextBox
LMG2L["TextBox_12"] = Instance.new("TextBox", LMG2L["ScrollingFrame_11"]);
LMG2L["TextBox_12"]["LineHeight"] = 2;
LMG2L["TextBox_12"]["TextXAlignment"] = Enum.TextXAlignment.Left;
LMG2L["TextBox_12"]["BorderSizePixel"] = 0;
LMG2L["TextBox_12"]["TextSize"] = 10;
LMG2L["TextBox_12"]["TextYAlignment"] = Enum.TextYAlignment.Top;
LMG2L["TextBox_12"]["BackgroundColor3"] = Color3.fromRGB(232, 232, 232);
LMG2L["TextBox_12"]["MultiLine"] = true;
LMG2L["TextBox_12"]["ClearTextOnFocus"] = false;
LMG2L["TextBox_12"]["Size"] = UDim2.new(0, 322, 0, 706);
LMG2L["TextBox_12"]["Position"] = UDim2.new(0, 12, 0, 0);
LMG2L["TextBox_12"]["Text"] = [[llll]];


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextButton
LMG2L["TextButton_13"] = Instance.new("TextButton", LMG2L["Frame_c"]);
LMG2L["TextButton_13"]["BorderSizePixel"] = 2;
LMG2L["TextButton_13"]["BackgroundColor3"] = Color3.fromRGB(174, 174, 174);
LMG2L["TextButton_13"]["Size"] = UDim2.new(0, 102, 0, 32);
LMG2L["TextButton_13"]["BorderColor3"] = Color3.fromRGB(104, 104, 104);
LMG2L["TextButton_13"]["Text"] = [[execute]];
LMG2L["TextButton_13"]["Position"] = UDim2.new(0, 6, 0, 264);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextButton
LMG2L["TextButton_14"] = Instance.new("TextButton", LMG2L["Frame_c"]);
LMG2L["TextButton_14"]["BorderSizePixel"] = 2;
LMG2L["TextButton_14"]["BackgroundColor3"] = Color3.fromRGB(174, 174, 174);
LMG2L["TextButton_14"]["Size"] = UDim2.new(0, 104, 0, 32);
LMG2L["TextButton_14"]["BorderColor3"] = Color3.fromRGB(104, 104, 104);
LMG2L["TextButton_14"]["Text"] = [[clear]];
LMG2L["TextButton_14"]["Position"] = UDim2.new(0, 114, 0, 264);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextButton
LMG2L["TextButton_15"] = Instance.new("TextButton", LMG2L["Frame_c"]);
LMG2L["TextButton_15"]["BorderSizePixel"] = 2;
LMG2L["TextButton_15"]["BackgroundColor3"] = Color3.fromRGB(174, 174, 174);
LMG2L["TextButton_15"]["Size"] = UDim2.new(0, 112, 0, 32);
LMG2L["TextButton_15"]["BorderColor3"] = Color3.fromRGB(104, 104, 104);
LMG2L["TextButton_15"]["Text"] = [[load]];
LMG2L["TextButton_15"]["Position"] = UDim2.new(0, 224, 0, 264);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.ScrollingFrame
LMG2L["ScrollingFrame_16"] = Instance.new("ScrollingFrame", LMG2L["Frame_c"]);
LMG2L["ScrollingFrame_16"]["ScrollingDirection"] = Enum.ScrollingDirection.Y;
LMG2L["ScrollingFrame_16"]["BorderSizePixel"] = 0;
LMG2L["ScrollingFrame_16"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["ScrollingFrame_16"]["Size"] = UDim2.new(0, 330, 0, 46);
LMG2L["ScrollingFrame_16"]["ScrollBarImageColor3"] = Color3.fromRGB(0, 0, 0);
LMG2L["ScrollingFrame_16"]["Position"] = UDim2.new(0, 6, 0, 302);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.ScrollingFrame
LMG2L["ScrollingFrame_17"] = Instance.new("ScrollingFrame", LMG2L["Frame_c"]);
LMG2L["ScrollingFrame_17"]["ScrollingDirection"] = Enum.ScrollingDirection.Y;
LMG2L["ScrollingFrame_17"]["BorderSizePixel"] = 0;
LMG2L["ScrollingFrame_17"]["BackgroundColor3"] = Color3.fromRGB(224, 224, 224);
LMG2L["ScrollingFrame_17"]["Size"] = UDim2.new(0, 158, 0, 296);
LMG2L["ScrollingFrame_17"]["ScrollBarImageColor3"] = Color3.fromRGB(0, 0, 0);
LMG2L["ScrollingFrame_17"]["Position"] = UDim2.new(0, 340, 0, 52);



local parent = LMG2L["ScrollingFrame_17"]

-- function สร้างปุ่ม
local function createButton(name, text, url)
    local btn = Instance.new("TextButton")
    btn.Name = name
    btn.Parent = parent
    btn.BorderSizePixel = 0
    btn.BackgroundColor3 = Color3.fromRGB(255,255,255)
    btn.Size = UDim2.new(0,140,0,30)
    btn.Text = text
    btn.TextWrapped = true

    btn.MouseButton1Click:Connect(function()
        local success, result = pcall(function()
            return game:HttpGet(url)
        end)

        if not success then
            warn("HttpGet failed:", url)
            return
        end

        local func, err = loadstring(result)
        if not func then
            warn("Loadstring error:", err)
            return
        end

        local ok, runtimeErr = pcall(func)
        if not ok then
            warn("Runtime error:", runtimeErr)
        end
    end)
end

-- 📦 รายการปุ่มทั้งหมด
local buttons = {
    {"TextButton9","project duck utg","https://raw.githubusercontent.com/ProBypasserHax1/Idkkk/refs/heads/main/projetduck.lua"},
    {"TextButton15","9/11 plane","https://pastebin.com/raw/5saVhNKb"},
    {"TextButton17","chara","https://pastebin.com/raw/g8vRx0wa"},
    {"TextButton7","SEGMA V4","https://pastefy.app/a2VxTbDs/raw"},
    {"TextButton18","dominant","https://raw.githubusercontent.com/kvsna3545-pixel/Dominant_executor/refs/heads/main/Main.luau"},
    {"TextButton2","OUCH SIMSOMP","https://pastefy.app/h6fhUC6z/raw"},
    {"TextButton4","project zaga","https://pastefy.app/rMyVKrO6/raw"},
    {"TextButton24","project two bfdi","https://pastebin.com/raw/3xHHRpsV"},
    {"TextButton8","project eluaria","https://pastefy.app/TvLHaW8s/raw"},
    {"TextButton21","project stigma","https://pastefy.app/QcuBoc7w/raw"},
    {"TextButton12","1×1×1×1×1hax gui v2","https://pastefy.app/MYWHQPCX/raw?part=1x1x1x1x1_HAX%20Gui%20V2"},
    {"TextButton23","project sligma","https://pastefy.app/QcuBoc7w/raw"},
    {"TextButton16","giant size destroyer","https://pastebin.com/raw/GqfUpdX6"},
    {"TextButton20","sigma gui private","https://pastebin.com/raw/78fUTxZV"},
    {"TextButton11","poongui by redkid","https://pastefy.app/LH5mdWwm/raw"},
    {"TextButton6","project segma v2","https://pastefy.app/EUwYT4gN/raw"},
    {"TextButton10","ss and febypass","https://pastebin.com/raw/rLZTaE8s"},
    {"TextButton14","EXTERNAL 1!1!","https://pastefy.app/4GLMKcft/raw"},
    {"TextButton5","private 1×1 v7.1.0","https://pastefy.app/tsPrm5Ky/raw"},
    {"TextButton3","SUMSUNG NOTE","https://pastefy.app/dA68xt1p/raw"},
    {"TextButton19","rc7 2016","https://fromhatai.neocities.org/Private/Rc7.lua"},
    {"TextButton","1×1 max mini gui","https://pastefy.app/4U9m0vuN/raw?part=1x1x1x1x1_hax%20Gui%20mini%20edition"},
    {"TextButton13","mang0 gui","https://pastebin.com/raw/6zDKS2TV"},
    {"TextButton22","shiba old","https://pastebin.com/raw/HrE2dKSD"},
}

-- 🔁 สร้างปุ่มทั้งหมด
for _, data in ipairs(buttons) do
    createButton(unpack(data))
end

-- 📐 layout
local layout = Instance.new("UIListLayout")
layout.Parent = parent

-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextButton
LMG2L["TextButton_32"] = Instance.new("TextButton", LMG2L["Frame_c"]);
LMG2L["TextButton_32"]["TextWrapped"] = true;
LMG2L["TextButton_32"]["BorderSizePixel"] = 0;
LMG2L["TextButton_32"]["TextStrokeColor3"] = Color3.fromRGB(255, 0, 0);
LMG2L["TextButton_32"]["TextScaled"] = true;
LMG2L["TextButton_32"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextButton_32"]["Size"] = UDim2.new(0, 46, 0, 24);
LMG2L["TextButton_32"]["Text"] = [[×]];
LMG2L["TextButton_32"]["Position"] = UDim2.new(0, 454, 0, 0);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.Frame.TextLabel
LMG2L["TextLabel_33"] = Instance.new("TextLabel", LMG2L["Frame_c"]);
LMG2L["TextLabel_33"]["BorderSizePixel"] = 0;
LMG2L["TextLabel_33"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextLabel_33"]["TextColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextLabel_33"]["BackgroundTransparency"] = 1;
LMG2L["TextLabel_33"]["Size"] = UDim2.new(0, 156, 0, 14);
LMG2L["TextLabel_33"]["Text"] = [[script hub]];
LMG2L["TextLabel_33"]["Position"] = UDim2.new(0, 340, 0, 38);


-- Players.giruoh_jgry.PlayerGui.ScreenGui.Frame key.TextButton
LMG2L["TextButton_34"] = Instance.new("TextButton", LMG2L["Frame key_2"]);
LMG2L["TextButton_34"]["TextWrapped"] = true;
LMG2L["TextButton_34"]["BorderSizePixel"] = 0;
LMG2L["TextButton_34"]["TextStrokeColor3"] = Color3.fromRGB(255, 0, 0);
LMG2L["TextButton_34"]["TextScaled"] = true;
LMG2L["TextButton_34"]["BackgroundColor3"] = Color3.fromRGB(255, 255, 255);
LMG2L["TextButton_34"]["Size"] = UDim2.new(0, 46, 0, 24);
LMG2L["TextButton_34"]["Text"] = [[×]];
LMG2L["TextButton_34"]["Position"] = UDim2.new(0, 454, 0, 0);

-- ========================
-- 🔧 SETUP
-- ========================
local frame = LMG2L["Frame key_2"]
local mainFrame = LMG2L["Frame_c"]

local textbox = LMG2L["TextBox_5"]
local enterBtn = LMG2L["TextButton_6"]
local clearBtn = LMG2L["TextButton_7"]
local closeBtn = LMG2L["TextButton_34"]

local scriptBox = LMG2L["TextBox_12"]
local executeBtn = LMG2L["TextButton_13"]
local clearScriptBtn = LMG2L["TextButton_14"]

-- 🔑 KEY
local correctKey = "00script"


-- ========================
-- 🧠 HELPER (ตัวช่วย)
-- ========================
local function notify(msg)
    print("[GUI]: " .. msg)
end

local function safeRun(func)
    local success, err = pcall(func)
    if not success then
        warn("Error: " .. err)
    end
end


-- ========================
-- 🔐 KEY SYSTEM
-- ========================
local function checkKey()
    if textbox.Text == correctKey then
        frame.Visible = false
        mainFrame.Visible = true
        notify("Login success")
    else
        textbox.Text = ""
        textbox.PlaceholderText = "wrong key!"
        notify("Wrong key")
    end
end


-- ========================
-- 🧼 CLEAR INPUT
-- ========================
local function clearText()
    textbox.Text = ""
end


-- ========================
-- ❌ CLOSE UI
-- ========================
local function closeUI()
    frame.Parent.Enabled = false
end


-- ========================
-- ⚡ EXECUTE SCRIPT
-- ========================
local function runScript()
    local code = scriptBox.Text

    safeRun(function()
        loadstring(code)()
    end)
end


-- ========================
-- 🧹 CLEAR SCRIPT BOX
-- ========================
local function clearScript()
    scriptBox.Text = ""
end


-- ========================
-- 🔗 CONNECT EVENTS
-- ========================
enterBtn.MouseButton1Click:Connect(checkKey)
clearBtn.MouseButton1Click:Connect(clearText)
closeBtn.MouseButton1Click:Connect(closeUI)

executeBtn.MouseButton1Click:Connect(runScript)
clearScriptBtn.MouseButton1Click:Connect(clearScript)


-- ========================
-- ⌨️ ENTER KEY SUPPORT
-- ========================
textbox.FocusLost:Connect(function(enterPressed)
    if enterPressed then
        checkKey()
    end
end)



return LMG2L["ScreenGui_1"], require;
